models/classes/Translation/Service/TranslationPostCreationService.php

```diff
@@ -0,0 +1,162 @@
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\Translation\Service;
+
+use core_kernel_classes_Resource;
+use core_kernel_persistence_Exception;
+use oat\generis\model\data\Ontology;
+use oat\tao\model\TaoOntology;
+use oat\tao\model\Translation\Entity\ResourceTranslation;
+use oat\tao\model\Translation\Exception\ResourceTranslationException;
+use oat\tao\model\Translation\Query\ResourceTranslationQuery;
+use oat\tao\model\Translation\Repository\ResourceTranslationRepository;
+use oat\taoTests\models\TaoTestOntology;
+use Psr\Log\LoggerInterface;
+use taoQtiTest_models_classes_QtiTestService;
+use Throwable;
+
+class TranslationPostCreationService
+{
+    private taoQtiTest_models_classes_QtiTestService $testQtiService;
+    private Ontology $ontology;
+    private ResourceTranslationRepository $resourceTranslationRepository;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        taoQtiTest_models_classes_QtiTestService $testQtiService,
+        Ontology $ontology,
+        ResourceTranslationRepository $resourceTranslationRepository,
+        LoggerInterface $logger
+    ) {
+        $this->testQtiService = $testQtiService;
+        $this->ontology = $ontology;
+        $this->resourceTranslationRepository = $resourceTranslationRepository;
+        $this->logger = $logger;
+    }
+
+    public function __invoke(core_kernel_classes_Resource $test): core_kernel_classes_Resource
+    {
+        try {
+            $jsonTest = $this->testQtiService->getJsonTest($test);
+            $testData = json_decode($jsonTest, true, 512, JSON_THROW_ON_ERROR);
+
+            $uniqueIds = $this->collectItemUniqueIds($testData);
+            $translationUris = $this->getItemTranslationUris($test, $uniqueIds);
+
+            $this->replaceOriginalItemsWithTranslations($testData, $translationUris);
+
+            $this->testQtiService->saveJsonTest($test, json_encode($testData));
+            $this->updateTranslationCompletionStatus($test, $uniqueIds, $translationUris);
+
+            return $test;
+        } catch (Throwable $exception) {
+            $this->logger->error('An error occurred during test translation: ' . $exception->getMessage());
```

Comment:
I would add the exception name here also
```suggestion
            $this->logger->error(
                sprintf(
                    'An error occurred during test translation: (%s) %s', 
                    get_class($exception),
                    $exception->getMessage()
                )
            );
```

Reply:
Done